src/unstract/sdk/x2txt.py

```diff
@@ -92,3 +109,32 @@ def get_x2text(self, adapter_instance_id: str) -> X2TextAdapter:
             self._adapter_instance_id = adapter_instance_id
             self._initialise()
         return self._x2text_instance
+
+    def push_usage_details(self, input_file_path: str, mime_type: str) -> None:
+        file_name = os.path.basename(input_file_path)
+
+        file_size = ToolUtils.get_file_size(input_file_path)
+
+        if mime_type == MimeType.PDF:
+            with pdfplumber.open(input_file_path) as pdf:
+                # calculate the number of pages
+                max_pages = len(pdf.pages)
+            Audit().push_page_usage_data(
+                platform_api_key=self._tool.get_env_or_die(ToolEnv.PLATFORM_API_KEY),
+                file_name=file_name,
+                file_size=file_size,
+                file_type=mime_type,
+                page_count=max_pages,
```

Comment:
@johnyrahul What if the pages to extract is explicitly mentioned in the LLM whisperer adapter? Always considering max_pages here won't work.

Reply:
Thanks @Deepak-Kesavan  for pointing it out, Addressed the change and tested.